pulsar-client-tools/src/main/java/org/apache/pulsar/admin/cli/CmdFunctions.java

```diff
@@ -468,9 +468,6 @@ private void inferMissingArguments(FunctionConfig functionConfig) {
             if (StringUtils.isEmpty(functionConfig.getNamespace())) {
                 inferMissingNamespace(functionConfig);
             }
-            if (StringUtils.isEmpty(functionConfig.getOutput())) {
```

Comment:
I would rather have this functionality gated by some kind of flag with default being output topic created that the users can turn off for optimization purposes

Reply:
but if user doesn't provide output topic name then it means user is not intended to publish output in the external topic. I think here, adding a flag creates a confusion which is not known by the user. So, as a user, I feel it's better if cli doesn't compute output topic if it's not provided.

Reply:
When we made this flag optional, the intention was that the user need to specify as few arguments as necessary and the system takes care of most of the arguments. Essentially design for user simplicity. That's why I felt if one needs an optimization, that might be a flag.
I'm curious what @sijie @jerrypeng feel about this.

Reply:
I don't think we should implicitly create a topic for the output.  I think it makes more since for the the user to explicitly specify it. The user might even be confused if he or she can submit a function without a output topic.  He or she might not even know what the default output topic name will be.  

If we really want to create a default output topic automatically for the function, lets at least check that the function is set to return a non-void type.  There is no point to create automatically creating an output topic if the function is set to return void

Reply:
> I don't think we should implicitly create a topic for the output.

can we merge this PR then. as I have mentioned into motivation: replication-namespace and security are the main concerns so, function should not create output topic implicitly.

Reply:
>  if people want to disable sink, there should be another flag controlling that.

I think when user submits function, user doesn't care about source or sink and as a user one
doesn't want to redirect output to any unknown topic. seems like adding extra flag makes output topic mandatory. not sure if that's a good idea.

Reply:
>  if people want to disable sink, there should be another flag controlling that.

@sijie , made the change.